File: daily/2025/3446_Sort_Matrix_by_Diagonals.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<vector<int>> sortMatrix(vector<vector<int>>& grid) {
        int n = grid.size();
        for (int r = 0; r < n; r++){
            vector<int> temp = {};
            int i = r;
            int j = 0;
            while(i < n) temp.push_back(grid[i++][j++]);
            sort(temp.begin(), temp.end(), greater<int>());
            i = r;
            j = 0;
            while(i < n) grid[i++][j++] = temp[j];
        }
        for (int c = 1; c < n; c++){
            vector<int> temp = {};
            int i = 0;
            int j = c;
            while(j < n) temp.push_back(grid[i++][j++]);
            sort(temp.begin(), temp.end());
            i = 0;
            j = c;
            while(j < n) grid[i++][j++] = temp[i];
        }
        return grid;
    }
};

int main(){
    vector<vector<int>> grid = {{1,7,3},{9,8,2},{4,5,6}};
    // vector<vector<int>> grid = {{0,1},{1,2}};
    // vector<vector<int>> grid = {{1}};
    Solution S;
    S.sortMatrix(grid);
    for (auto row : grid){
        for (auto element : row){
            cout << element << " ";
        }
        cout << endl;
    }
    return 0;
}

File: daily_python/3446_Sort_Matrix_by_Diagonals.py
import os
import sys
from typing import List

class Solution:
    def sortMatrix(self, grid: List[List[int]]) -> List[List[int]]:
        n = len(grid)
        for row in range(n):
            temp = [grid[row + k][k] for k in range(n - row)]
            temp = sorted(temp, reverse = True)
            for k in range(n - row):
                grid[row + k][k] = temp[k]
        for col in range(1, n):
            temp = [grid[k][col + k] for k in range(n - col)]
            temp = sorted(temp)
            for k in range(n - col):
                grid[k][col + k] = temp[k]
        return grid

def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            grid = [list(map(int, row.split(','))) for row in lines[0].split('=')[1].strip()[2:-3].split('},{')]
            dataset.append(grid)
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for grid in dataset:
        results.append(solution.sortMatrix(grid))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')